fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneOnlineIndexingTest.java

```diff
@@ -1344,5 +1351,118 @@ void luceneOnlineIndexingTestTerribleRebalance2ndChance() {
             assertTrue(Terriblerebalnce2ndChanceIndexMaintainer.isSecondPass);
         }
     }
+
+
+    @Test
+    void testConflictWithLock() throws IOException, InterruptedException {
```

Comment:
This test has a bunch of moving pieces, for posterity it would probably be good to comment exactly what is trying to happen here, something like:
> If the file lock gets a conflict while closing the lock, it should ensure that the lock is released either during it's close, or the directory's close so that future operations can continue.

Reply:
Probably also worth adding some similar context to the Issue #2731 which currently has no description.

Reply:
✅